examples/example.c

```diff
@@ -28,6 +28,17 @@
 #    define sleep_s(SECONDS) sleep(SECONDS)
 #endif
 
+static double
+traces_sampler_callback(sentry_sampling_context_t *sampling_ctx)
+{
+    if (sentry_value_as_int32(sentry_value_get_by_key(
+            sentry_sampling_context_get_custom_context(sampling_ctx), "b"))
```

Comment:
We are currently passing in this sampling context consisting of custom context (`sentry_value_t`) and a transaction context (`sentry_transaction_context_t`), but the latter is not all that useful as we can't get any data from it.

 We have the following setters `transaction_context_set_name` & `transaction_context_set_operation` but no getter functions (e.g. for these `name`/`operation` values or `trace_id`/`parent_span_id` when updating from header).

I'm not sure if adding access to transaction context data should be part of this PR (or even happen in the first place), but it does feel like something worth considering.

Reply:
It should exist because the transaction context would otherwise be useless inside the callback. The question is how much of it should be exposed and whether we provide specialized accessors for each item or a getter to access the inner.

@markushi, what are typical things people filter on? Does providing an interface to a subset make sense, or should we expose the whole inner value?

I would add this to this PR and would prefer to provide access to the inner value, which we can freeze before returning.